### DISCUSSION TITLE
[Wayland] Can Qtile identify itself as the WM?

### DISCUSSION THREAD

**Discussion by grtcdr** [qtile/qtile#2998] [Ideas] [answered]
I work on a system information fetcher that can display window manager information along with the session. And my question is, as a developer, how can I tell if Qtile (on Wayland) is the _running window manager_ and display that to my users?

For example: Sway uses a UNIX socket (for some IPC communication?), and exposes an environment variable (SWAYSOCK) which always points to a real file that exists on the filesystem, allowing us to know that Sway is the running WM.

**Comment by m-col** [comment, marked as answer]
We also have an IPC socket that you can test for which can be found at `${XDG_CACHE_HOME:-~/.cache}/qtile/qtilesocket.$WAYLAND_DISPLAY`. Under X11 it's the same but with `$DISPLAY` at the end.

**Reply by grtcdr** [reply]
Thank you :)

**Comment by RedBearAK** [comment]
@m-col 

Almost three years later, is there any specific reason Qtile doesn't set helpful environment variables like `XDG_SESSION_DESKTOP` and/or `XDG_CURRENT_DESKTOP`? I've been using those with great success on quite a few desktop environments and window managers before someone tested my detection script on Qtile and found both variables were unset.

**Reply by RedBearAK** [reply]
Might have answered my own question, to an extent. User `systemd` services have trouble accessing a lot of the user's environment variables without help, and I've been using a `systemd import-environment` command in a desktop entry file in `~/.config/autostart/` to overcome that problem. 

Window managers that don't run that autostart item end up appearing to not have a lot of environment variables set. 

But, a Qtile user showed me the results of echoing those variable names and they were also empty in the terminal, so maybe there is still a problem with Qtile not setting those variables at login.

**Reply by nomisreual** [reply]
Noticed the same after trying out microfetch running NixOs
ended up setting these two envs myself for now
